core/src/main/java/com/linecorp/armeria/client/DecoratingClientFactory.java

```diff
@@ -49,10 +49,28 @@ protected DecoratingClientFactory(ClientFactory delegate) {
     /**
      * Returns the delegate {@link ClientFactory}.
      */
-    protected ClientFactory delegate() {
+    protected final ClientFactory delegate() {
         return delegate;
     }
 
+    /**
+     * Creates a new {@link HttpClient} which uses the same {@link SessionProtocol}, {@link EndpointGroup} and
+     * {@link ClientOptions} with the specified {@link ClientBuilderParams}. Note that {@code path} and
+     * {@link SerializationFormat} are always {@code "/"} and {@link SerializationFormat#NONE}.
+     */
+    protected final HttpClient newHttpClientDelegate(ClientBuilderParams params) {
+        final URI uri = params.uri();
+        if (Clients.isUndefinedUri(uri)) {
+            return (HttpClient) delegate().newClient(
```

Comment:
I think `delegate()` is correct because a `DecoratingClientFactory` usually has `HttpClientFactory` as its delegate. It's not very clear, though. Let me make this method require a `ClientFactory` explicitly.